### PR TITLE
chore: add info about aad-pod-identity deprecation

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* https://github.com/kedacore/keda-docs/pull/1235
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to #5035

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]

**Comment by JorTurFer** [comment]
@tomkerkhove ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ You can find all deprecations in [this overview](https://github.com/kedacore/ked
 
 New deprecation(s):
 
-- TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- Removal of AAD-Pod-Identity ([#5035](https://github.com/kedacore/keda/issues/5035))
```
Comment (tomkerkhove) [inline CHANGELOG.md:74]:
```suggestion
- Remove support for AAD Pod Identity-based authentication ([#5035](https://github.com/kedacore/keda/issues/5035))
```

**Review by tomkerkhove — APPROVED** [review]

File: pkg/scaling/resolver/scale_resolvers.go [hunk 2]
```diff
@@ -198,6 +198,10 @@ func ResolveAuthRefAndPodIdentity(ctx context.Context, client client.Client, log
 		case kedav1alpha1.PodIdentityProviderAwsKiam:
 			authParams["awsRoleArn"] = podTemplateSpec.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
 		case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
+			if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAzure {
+				// FIXME: Delete this for v2.15
+				logger.Info("WARNING: aad-pod-identity is out of support (https://github.com/Azure/aad-pod-identity#-announcement) and it'll be removed from KEDA on v2.15")
```
Comment (tomkerkhove) [inline pkg/scaling/resolver/scale_resolvers.go:203]:
```suggestion
				logger.Info("WARNING: Azure AD Pod Identity is out of support (https://github.com/Azure/aad-pod-identity#-announcement) and it'll be removed from KEDA on v2.15")
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -71,7 +71,7 @@ You can find all deprecations in [this overview](https://github.com/kedacore/ked
 
 New deprecation(s):
 
-- TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- Remove support for AAD Pod Identity-based authentication ([#5035](https://github.com/kedacore/keda/issues/5035))
```
Comment (tomkerkhove) [inline CHANGELOG.md:74]:
```suggestion
- Remove support for Azure AD Pod Identity-based authentication ([#5035](https://github.com/kedacore/keda/issues/5035))
```

**Pull request merged**